src/providers/node.rs

```diff
@@ -367,6 +437,59 @@ impl NodeProvider {
 
         all_deps
     }
+
+    pub fn is_nx_monorepo(app: &App) -> bool {
+        app.includes_file("nx.json")
+    }
+
+    pub fn get_nx_app_name(app: &App, env: &Environment) -> Result<Option<String>> {
+        if let Some(app_name) = env.get_config_variable(NX_APP_NAME_ENV_VAR) {
+            return Ok(Some(app_name));
+        }
+
+        if let Ok(nx_json) = app.read_json::<NxJson>("nx.json") {
+            if let Some(default_project) = nx_json.default_project {
+                return Ok(Some(default_project.as_str().unwrap().to_owned()));
+            }
+        }
+
+        return Err(anyhow!("Could not derive nx app to build and run. Please add a default project to your nx config or set NIXPACKS_{}", NX_APP_NAME_ENV_VAR));
+    }
+
+    pub fn get_nx_project_json_for_app(app: &App, env: &Environment) -> Result<ProjectJson> {
+        let app_name = NodeProvider::get_nx_app_name(app, env)?.unwrap();
+        let project_path = format!("./apps/{}/project.json", app_name);
+        let nx_app_project_json = app.read_json::<ProjectJson>(&project_path);
+
+        if nx_app_project_json.is_err() {
+            return Err(anyhow!(format!(
```

Comment:
Correct me if I am wrong, but I don't think that a `project.json` file is required for all NX apps and the config for each app can be at the project root. In this case I think it makes more sense for this function to return `Result<Option<ProjectJson>>` instead of throwing an error.

Reply:
True! It would fail way before they even try to build with nixpacks.